src/annotator/util/geometry.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Return the intersection of two rects.
+ *
+ * @param {DOMRect} rectA
+ * @param {DOMRect} rectB
+ */
+export function intersectRects(rectA, rectB) {
+  const left = Math.max(rectA.left, rectB.left);
+  const right = Math.min(rectA.right, rectB.right);
+  const top = Math.max(rectA.top, rectB.top);
+  const bottom = Math.min(rectA.bottom, rectB.bottom);
+  return new DOMRect(left, top, right - left, bottom - top);
+}
+
+/**
+ * Return `true` if a rect is _empty_.
+ *
+ * An empty rect is defined as one with zero or negative width/height, eg.
+ * as returned by `new DOMRect()` or `Element.getBoundingClientRect()` for a
+ * hidden element.
+ *
+ * @param {DOMRect} rect
+ */
+export function rectIsEmpty(rect) {
+  return rect.width <= 0 || rect.height <= 0;
+}
+
+/**
+ * Return true if the lines a-b and c-d overlap. The inputs must be normalized
+ * such that b >= a and d >= c.
+ *
+ * @param {number} a
+ * @param {number} b
+ * @param {number} c
+ * @param {number} d
+ */
```

Comment:
I wonder if a little ASCII art could help illustrate what this does?

```suggestion
/**
 * Return true if the lines a-b and c-d overlap.
 * 
 * e.g. these lines overlap:
 *              a = 5 -------------- b = 10
 *       c = 3 ------------- d = 7
 *
 *  The inputs must be normalized such that b >= a and d >= c.
 * @param {number} a
 * @param {number} b
 * @param {number} c
 * @param {number} d
 */
```

Reply:
Good suggestion - will do.